sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -457,20 +531,45 @@ class CudfRegexTranspiler(mode: RegexMode) {
   // rejected by the transpiler
   private val nothingToRepeat = "nothing to repeat"
 
+  private def countCaptureGroups(regex: RegexAST): Int = {
+    regex match {
+      case RegexSequence(parts) => parts.foldLeft(0)((c, re) => c + countCaptureGroups(re))
+      case RegexGroup(capture, base) => 
+        if (capture) {
+          1 + countCaptureGroups(base)
+        } else {
+          countCaptureGroups(base)
+        }
+      case _ => 0
+    }
+  }
+
   /**
    * Parse Java regular expression and translate into cuDF regular expression.
    *
    * @param pattern Regular expression that is valid in Java's engine
    * @return Regular expression in cuDF format
    */
-  def transpile(pattern: String): String = {
+  def transpile(pattern: String, repl: Option[String]): (Option[String], Option[String]) = {
```

Comment:
I don't think we need the first item in the tuple to be optional since we always return a value